src/common/snippets/tests/src/lir_comparator.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "lir_comparator.hpp"
+
+#include "common_test_utils/graph_comparator.hpp"
+#include "snippets/lowered/pass/pass.hpp"
+#include "common_test_utils/common_utils.hpp"
+
+using namespace ov::snippets::lowered;
+
+namespace std {
+template <typename T>
+inline string to_string(const vector<T>& vec) {
+    return ov::test::utils::vec2str<T>(vec);
+}
+
+inline string to_string(const ov::snippets::Reg& reg) {
+    return string("Reg(type = " + ov::snippets::regTypeToStr(reg.type) + ", idx = " + to_string(reg.idx) + ")");
+}
+
+inline string to_string(const ov::Node::type_info_t& info) {
+    stringstream ss;
+    ss << info;
+    return ss.str();
+}
+} // namespace std
+
+namespace ov {
+namespace test {
+namespace snippets {
+#define COMPARE(error_msg, lhs, rhs) \
+    if (lhs != rhs)                  \
+        return Result::error(std::string(error_msg) + " are different: " + std::to_string(lhs) + " and " + std::to_string(rhs))
+
+#define PROPAGATE_ERROR(prefix, result) \
+    if (!result.valid)                  \
+        return Result::error(std::string(prefix) + ": " + result.message)
+
+LIRComparator::Result LIRComparator::compare(const std::shared_ptr<LinearIR>& linear_ir,
```

Comment:
Maybe I want so much 😄 But is it possible to verify `PortConnectors` between expression?
For example, need to check that `MatMul` has 2 consumers and `MatMul_Ref` has also 2 consumers

Maybe not at the moment, as follow-up task to improve comparator. What do you think?

Reply:
I added `compare_port_connectors` method, please check.
If you have further suggestions on what can be checked in LIR comparator, feel free to share them here or in the ticket 139926